grammars/level16-Additions.lark

```diff
@@ -3,5 +3,7 @@
 ?atom: NUMBER | var | list_access | unsupported_number
 list_access: var _LEFT_SQUARE_BRACKET (INT | random | var) _RIGHT_SQUARE_BRACKET
 change_list_item: var _LEFT_SQUARE_BRACKET (INT | var) _RIGHT_SQUARE_BRACKET _SPACE _EQUALS _SPACE (var | textwithoutspaces)
-list_access_var: var _SPACE _EQUALS _SPACE var _LEFT_SQUARE_BRACKET (INT | random | var) _RIGHT_SQUARE_BRACKET
-assign_list: var _SPACE _IS _SPACE _LEFT_SQUARE_BRACKET (quoted_text | NUMBER) (_COMMA _SPACE? (quoted_text | NUMBER))+ _RIGHT_SQUARE_BRACKET
+list_access_var: var _SPACE (_EQUALS|_IS) _SPACE var _LEFT_SQUARE_BRACKET (INT | random | var) _RIGHT_SQUARE_BRACKET
+assign_list: var _SPACE (_EQUALS|_IS) _SPACE _LEFT_SQUARE_BRACKET (quoted_text | NUMBER) (_COMMA _SPACE? (quoted_text | NUMBER))+ _RIGHT_SQUARE_BRACKET
+
+# the two rules above now overlap :(
```

Comment:
They shouldn't be. `var` shouldn't match `[`. Just some wild guesses, but maybe `var: NAME -> var` or the complexity of the `ID_START` regex confuses lark about which rules overlap and which don't or cause it to hit some edge case in the lexer generator?

Reply:
yeah this was really a note to self so I can continue tomorrow, no help needed at this point